scene/gui/flow_container.cpp

```diff
@@ -0,0 +1,252 @@
+		if (!child || !child->is_visible_in_tree()) {
+			continue;
+		}
+		if (child->is_set_as_top_level()) {
+			continue;
+		}
+		Size2i child_size = children_minsize_cache[child];
+
+		_LineData line_data = lines_data[current_line_idx];
+		if (child_idx_in_line >= lines_data[current_line_idx].child_count) {
+			current_line_idx++;
+			child_idx_in_line = 0;
+			if (vertical) {
+				ofs.x += line_data.min_line_height + separation_horizontal;
+				ofs.y = 0;
+			} else {
+				ofs.x = 0;
+				ofs.y += line_data.min_line_height + separation_vertical;
+			}
+			line_data = lines_data[current_line_idx];
+		}
+
+		if (vertical) { /* VERTICAL */
+			if (child->get_h_size_flags() & (SIZE_FILL | SIZE_SHRINK_CENTER | SIZE_SHRINK_END)) {
+				child_size.width = line_data.min_line_height;
+			}
+
+			if (child->get_v_size_flags() & SIZE_EXPAND) {
+				int stretch = line_data.stretch_avail * child->get_stretch_ratio() / line_data.stretch_ratio_total;
+				child_size.height += stretch;
+			}
+
+		} else { /* HORIZONTAL */
+			if (child->get_v_size_flags() & (SIZE_FILL | SIZE_SHRINK_CENTER | SIZE_SHRINK_END)) {
+				child_size.height = line_data.min_line_height;
+			}
+
+			if (child->get_h_size_flags() & SIZE_EXPAND) {
+				int stretch = line_data.stretch_avail * child->get_stretch_ratio() / line_data.stretch_ratio_total;
+				child_size.width += stretch;
+			}
+		}
+
+		Rect2 child_rect = Rect2(ofs, child_size);
+		if (rtl) {
+			child_rect.position.x = get_rect().size.x - child_rect.position.x - child_rect.size.width;
+		}
+
+		fit_child_in_rect(child, child_rect);
+
+		if (vertical) { /* VERTICAL */
+			ofs.y += child_size.height + separation_vertical;
+		} else { /* HORIZONTAL */
+			ofs.x += child_size.width + separation_horizontal;
+		}
+
+		child_idx_in_line++;
+	}
+	cached_size = (vertical ? ofs.x : ofs.y) + line_height;
+	cached_line_count = current_line_idx;
+}
+
+Size2 FlowContainer::get_minimum_size() const {
+	Size2i minimum;
+
+	for (int i = 0; i < get_child_count(); i++) {
+		Control *c = Object::cast_to<Control>(get_child(i));
+		if (!c) {
+			continue;
+		}
+		if (c->is_set_as_top_level()) {
+			continue;
+		}
+
+		if (!c->is_visible()) {
+			continue;
+		}
+
+		Size2i size = c->get_combined_minimum_size();
+
+		if (vertical) { /* VERTICAL */
+			minimum.height = MAX(minimum.height, size.height);
+			minimum.width = cached_size;
+
+		} else { /* HORIZONTAL */
+			minimum.width = MAX(minimum.width, size.width);
+			minimum.height = cached_size;
+		}
+	}
+
+	return minimum;
+}
+
+void FlowContainer::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_SORT_CHILDREN: {
+			_resort();
+			update_minimum_size();
+		} break;
+		case NOTIFICATION_THEME_CHANGED: {
+			update_minimum_size();
+		} break;
+		case NOTIFICATION_TRANSLATION_CHANGED:
+		case NOTIFICATION_LAYOUT_DIRECTION_CHANGED: {
+			queue_sort();
+		} break;
+	}
+}
+
+int FlowContainer::get_line_count() const {
+	return cached_line_count;
```

Comment:
```suggestion
	return cached_line_count + 1;
```
It returns 0 if the container doesn't wrap.

Reply:
Fixed by setting `cached_line_count` via `lines_data.size()` in `_resort()`.